l3_dump.py

```diff
@@ -9,40 +9,198 @@
 import struct
 import subprocess
 import os
+import platform
+import shutil
+import re
 
 # ##############################################################################
 # Constants that tie the unpacking logic to L3's core structure's layout
 # ##############################################################################
 L3_LOG_HEADER_SZ = 32  # bytes; offsetof(L3_LOG, slots)
 L3_ENTRY_SZ = 32       # bytes; sizeof(L3_ENTRY)
 
-if len(sys.argv) != 3:
-    print(f"Usage: {sys.argv[0]} <logfile> <program-binary>")
-    print(" ")
-    print("Note: If L3 logging is done for <program-binary> with L3_LOC_ENABLED=1")
-    print("we expect to find the LOC-decoder binary named <program-binary>_loc")
-    sys.exit(0)
-
 # #############################################################################
-# Check for required LOC-decoder binary, based on environment of run
-# If L3_LOC_ENABLED=1 is set in the env, search for a LOC-decoder binary
-# named "<program_binary>_loc".
 # #############################################################################
-PROGRAM_BIN = sys.argv[2]
+PROGRAM_BIN = 'unknown'
 DECODE_LOC_ID = 0   # By default, we expect L3 logging was done w/LOC OFF.
 LOC_ENABLED = "L3_LOC_ENABLED"
 LOC_DECODER = "LOC_DECODER"
-if LOC_ENABLED in os.environ:
-    env_var_val = os.getenv(LOC_ENABLED)
-    if env_var_val == "1":
-        LOC_DECODER = PROGRAM_BIN + "_loc"
-        if os.path.exists(LOC_DECODER) is False:
-            print(f"Env-var {LOC_ENABLED}=1 is set, but required "
-                  + f"LOC-decoder binary {LOC_DECODER} is not found.")
-            sys.exit(1)
-
-        # LOC is in-use and LOC-decoder binary was found.
-        DECODE_LOC_ID = 1
+
+# #############################################################################
+# Establish path / names to tools used here based on the OS-platform version
+# #############################################################################
+OS_UNAME_S = platform.system()
+
+if OS_UNAME_S == 'Linux':
+    READELF_BIN = 'readelf'
+    READELF_HEXDUMP_ARG = '-x'
+    READELF_STRDUMP_ARG = '-p'
+    READELF_STRING_SEP = ']  '
+    READELF_DATA_SECTION = '.rodata'
```

Comment:
I quite don't like this inline "global" code -- this should also be in a function, but I could not figure out immediately the best way to do that now.

I laid it out like-this, so that imminent support for Mac/OSX can add its logic here.